.github/templates/main.yml

```diff
@@ -266,10 +276,40 @@ jobs:
         _: #@ template.replace(msbuild("Tests/Realm.Tests", TargetFramework="net461", RestoreConfigFile="${{ github.workspace }}/Tests/Test.NuGet.Config", UseRealmNupkgsWithVersion="${{ needs.build-packages.outputs.package_version }}" ))
       - name: Run the tests
         run: #@ "${{ github.workspace }}/Tests/Realm.Tests/bin/" + configuration + "/net461/Realm.Tests.exe --result=TestResults.Windows.xml --labels=After"
-      - name: Publish Unit Test Results
-        uses: EnricoMi/publish-unit-test-result-action/composite@v1
-        if: always()
-        with:
-          files: TestResults.Windows.xml
-          comment_on_pr: false
-          check_name: Windows .NET framework tests results
+      - #@ template.replace(publishTestsResults("TestResults.Windows.xml", "Windows .NET framework tests results"))
+  run-tests-netcore-net5:
+    name: Run tests for .NetCore and .Net5
+    needs: build-packages
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macos-latest, windows-latest, ubuntu-latest ]
+        targetFramework: [ netcoreapp3.1, net5.0 ]
+    steps:
+    - #@ checkoutCode()
+    - #@ template.replace(fetchPackageArtifacts( [ "Realm", "Realm.Fody" ] ))
+    - name: Build the test
+      run: #@ "dotnet build ${{ github.workspace }}/Tests/Realm.Tests -c " + configuration + " -f ${{ matrix.targetFramework }} -p:RestoreConfigFile=${{ github.workspace }}/Tests/Test.NuGet.Config -p:UseRealmNupkgsWithVersion=${{ needs.build-packages.outputs.package_version }} -p:AddNet5Framework=${{ matrix.targetFramework == 'net5.0' && 'true' || 'false' }}"
+    - name: Run the tests
+      run: #@ "dotnet run -p ${{ github.workspace }}/Tests/Realm.Tests -c " + configuration + " -f ${{ matrix.targetFramework }} --no-build -- --labels=After --result=${{ github.workspace }}/TestResults.NetCore.xml"
```

Comment:
Same for dotnet run?

Reply:
Also, since we already built the binary, can we just run it rather than use `dotnet run`? It's a bit more hassle finding the right location, but it will make it easier to debug things in case of crashes. I had to do this here: https://github.com/realm/realm-dotnet/pull/2355/files.